K2Field.K2NE.ServiceBroker/ServiceObjects/EmailTemplate/Template.cs

```diff
@@ -0,0 +1,171 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Text;
+using System.Text.RegularExpressions;
+using SourceCode.SmartObjects.Client;
+
+namespace K2Field.K2NE.ServiceBroker.ServiceObjects.EmailTemplate
+{
+    /// <summary>
+    /// The Class is used to work with EmailTemplates. 
```

Comment:
Actually, this class does not do any translation. It is only used for recursion and to find a template inside a template. So in the end it returns Subject and Body with any found %Template.Name% placeholder replaced with a certain template. The translation is done later in the EmailTemplateSO. My main idea was to keep only 1 connection open and to close it as soon as I do not need any data from smo. That is why I am using 2 constructors - one for Read method and another one for List method. But in general the implementation is the same. I pass into the SMOClientServer connection and it does all the work inside.